ilastik/experimental/api/_base.py

```diff
@@ -59,13 +59,28 @@ def predict(self, data):
                 )
 
             self._reorder_op.Input.setValue(data)
-
-            data = self._predict_op.PMaps.value[...]
+            roi = xarray_roi_to_slicing(xarray_roi, self._predict_op.PMaps.meta.axistags.keys())
+            data = self._predict_op.PMaps[roi].wait()
             return xarray.DataArray(data, dims=tuple(self._predict_op.PMaps.meta.axistags.keys()))
 
     return _PipelineImpl()
 
 
+def xarray_roi_to_slicing(xarray_roi, vigra_axistags):
+    if any(k not in vigra_axistags for k in xarray_roi.keys()):
+        raise ValueError(f"xarray_roi contains keys ({xarray_roi.keys()} not in axistags ({vigra_axistags})")
+    if not xarray_roi:
```

Comment:
Do we really need this special case? That is, does `a[...] == a[:, :, :, :, :]` always holds in volumina?

Reply:
it depends on the data, unfortunately. I wanted to send a clear message for what happens, when no roi is provided.